src/test/java/com/uid2/admin/vertx/EncryptionKeyServiceTest.java

```diff
@@ -177,21 +172,41 @@ void addSiteKeyUsesKeysetAndAddsKey() throws Exception {
     }
 
     @Test
-    void addKeysetKey() throws Exception {
+    void whenAddFirstKeyForKeysetItIsImmediatelyActiveAndUploaded() throws Exception {
         setKeysetKeys(123);
-        Map<Integer, AdminKeyset> keysets = new HashMap<Integer, AdminKeyset>() {{
+        Map<Integer, AdminKeyset> keysets = new HashMap<>() {{
             put(1, new AdminKeyset(1, 5, "test", Set.of(4,6,7), Instant.now().getEpochSecond(),true, true, new HashSet<>()));
         }};
         setAdminKeysets(keysets);
         final KeysetKey key = keyService.addKeysetKey(1);
         verify(keysetKeyStoreWriter).upload(collectionOfSize(1), eq(124));
-        assertSiteKeyActivation(key, clock.now());
+
+        final int siteKeyActivatesInSeconds = 0; //since this is the first key for a keyset, test that it is immediately active
+        assertKeyActivation(clock.now(), siteKeyActivatesInSeconds, SITE_KEY_EXPIRES_AFTER_SECONDS, key.getCreated(), key.getActivates(), key.getExpires());
     }
 
+    @Test
+    void whenAddSecondKeyForKeysetItIsNotImmediatelyActive() throws Exception {
```

Comment:
Nit pick, do we want to have one test case that when `isDuringRotation` is true?

Reply:
There are lots of existing rotation checks below. These call `checkRotatedKeyResponse` which ensures `activates` has the correct timestamp.